qoffer-infrastructure/src/main/groovy/life/qbic/portal/qoffer2/web/views/create/offer/SelectItemsView.groovy

```diff
@@ -75,53 +73,59 @@ class SelectItemsView extends VerticalLayout{
-
-        initLayout()
-        setupDataProvider()
-        addListener()
-    }
-
-    /**
-     * Listens for changes of the foundProducts list and creates ProductItemViewModel objects from them
-     */
-    def listener = {
-        if (it instanceof ObservableList.ElementEvent)  {
-            foundProducts.each {
-                if(it instanceof Sequencing){
-                    sequencingProduct.add(new ProductItemViewModel(0, it))
+        sequencingProduct = createOfferViewModel.sequencingProducts as ObservableList
+        sequencingProduct.addPropertyChangeListener({
+            if (it instanceof ObservableList.ElementEvent) {
+                sequencingProduct.each {
                     sequencingGrid.dataProvider.refreshAll()
                 }
-                else if(it instanceof ProjectManagement){
-                    projectManagementProduct.add(new ProductItemViewModel(0, it))
+            }
+        })
+
+        projectManagementProduct = createOfferViewModel.managementProducts as ObservableList
+        projectManagementProduct.addPropertyChangeListener({
+            if (it instanceof ObservableList.ElementEvent) {
+                projectManagementProduct.each {
                     projectManagementGrid.dataProvider.refreshAll()
                 }
-                else if(it instanceof PrimaryAnalysis){
-                    primaryAnalyseProduct.add(new ProductItemViewModel(0, it))
+            }
+        })
+
+        storageProduct = createOfferViewModel.storageProducts as ObservableList
+        storageProduct.addPropertyChangeListener({
+            if (it instanceof ObservableList.ElementEvent) {
+                storageProduct.each {
+                    storageGrid.dataProvider.refreshAll()
+                }
+            }
+        })
+
+        primaryAnalyseProduct = createOfferViewModel.primaryAnalysisProducts as ObservableList
+        primaryAnalyseProduct.addPropertyChangeListener({
+            if (it instanceof ObservableList.ElementEvent) {
+                primaryAnalyseProduct.each {
                     primaryAnalyseGrid.dataProvider.refreshAll()
                 }
-                else if(it instanceof SecondaryAnalysis){
-                    secondaryAnalyseProduct.add(new ProductItemViewModel(0, it))
+            }
+        })
+
+        secondaryAnalyseProduct = createOfferViewModel.secondaryAnalysisProducts as ObservableList
+        secondaryAnalyseProduct.addPropertyChangeListener({
+            if (it instanceof ObservableList.ElementEvent) {
+                secondaryAnalyseProduct.each {
                     secondaryAnalyseGrid.dataProvider.refreshAll()
                 }
-                else if(it instanceof DataStorage){
-                    storageProduct.add(new ProductItemViewModel(0, it))
-                    storageGrid.dataProvider.refreshAll()
-                }
             }
-        }
-    } as PropertyChangeListener
+        })
+
+        this.selectedItems = []
```

Comment:
Why not use the viewmodel for the selected items?

Reply:
this will be addressed in another PR